src/backend/access/heap/heapam.c

```diff
@@ -2195,7 +2195,7 @@ heap_insert(Relation relation, HeapTuple tup, CommandId cid,
 	 * the heaptup data structure is all in local memory, not in the shared
 	 * buffer.
 	 */
-	CacheInvalidateHeapTuple(relation, heaptup, NULL);
+	CacheInvalidateHeapTuple(relation, heaptup, NULL, false);
```

Comment:
this change is specifically for perf improvement. We should have data in commit description clearly stating test case and improvement achieved via this change. We can have a multi connection test where all the connections run a procedure with table variables.

Reply:
Updated original comment with some test results. I will format and publish all numbers from this experiment in a separate document, thought it was best to concisely summarize results in the commit description. 

As an additional note, JDBC tests pass after running.